compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/PromoteUsedTemporaries.ts

```diff
@@ -161,6 +190,7 @@ export function promoteUsedTemporaries(fn: ReactiveFunction): void {
     }
   }
   visitReactiveFunction(fn, new Visitor(), state);
+  visitReactiveFunction(fn, new Visitor2(), state);
```

Comment:
this needs a comment but the passes are now:
* collect promotable temporaries
* promote temporaries
* find any other Identifier instances with the same DeclarationId as something we promoted, and promote them too